src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -346,26 +340,66 @@ def _update_params_for_recommendation_id(
         recommendation_job_name = inference_recommendation_id.split("/")[0]
 
         sage_client = self.sagemaker_session.sagemaker_client
-        recommendation_res = sage_client.describe_inference_recommendations_job(
-            JobName=recommendation_job_name
-        )
-        input_config = recommendation_res["InputConfig"]
 
-        recommendation = next(
-            (
-                rec
-                for rec in recommendation_res["InferenceRecommendations"]
-                if rec["RecommendationId"] == inference_recommendation_id
-            ),
-            None,
-        )
+        # Retrieve model or inference recommendation job details
+        recommendation_res, model_res = None, None
+        try:
+            recommendation_res = sage_client.describe_inference_recommendations_job(
+                JobName=recommendation_job_name
+            )
+        except sage_client.exceptions.ResourceNotFound:
+            pass
+        try:
+            model_res = sage_client.describe_model(ModelName=recommendation_job_name)
+        except sage_client.exceptions.ResourceNotFound:
+            pass
+        if recommendation_res is None and model_res is None:
+            raise ValueError("Inference Recommendation id is not valid")
 
-        if not recommendation:
-            raise ValueError(
-                "inference_recommendation_id does not exist in InferenceRecommendations list"
+        # Search the recommendation from above describe result lists
+        inference_recommendation, instant_recommendation = None, None
+        if recommendation_res:
+            inference_recommendation = next(
+                (
+                    rec
+                    for rec in recommendation_res["InferenceRecommendations"]
+                    if rec["RecommendationId"] == inference_recommendation_id
+                ),
+                None,
             )
+        if model_res:
+            instant_recommendation = next(
+                (
+                    rec
+                    for rec in model_res["DeploymentRecommendation"][
+                        "RealTimeInferenceRecommendations"
+                    ]
+                    if rec["RecommendationId"] == inference_recommendation_id
+                ),
+                None,
+            )
+        if inference_recommendation is None and instant_recommendation is None:
+            raise ValueError("Inference Recommendation id does not exist")
+
+        # Update params beased on instant recommendation
+        if instant_recommendation:
```

Comment:
could we not call it "instant"?

maybe use the names "right_size_recommendation" and "model_recommendation"? 

Because that indicates the source of the recommendation

Reply:
makes sense, will rename them